src/auth.test.js

```diff
@@ -299,6 +302,39 @@ describe('auth', () => {
     })
   })
 
+  describe('linkWithPopup(authProvider)', () => {
+    it('returns link result', () => {
+      const authProvider = 'skqdk'
+      const credential = 'qosdqkds'
+      const iterator = authModule.linkWithPopup.call(context, authProvider)
+
+      expect(iterator.next().value)
+        .toEqual(call([auth.currentUser, auth.currentUser.linkWithPopup], authProvider))
+
+      expect(iterator.next({ credential })).toEqual({
+        done: true,
+        value: { credential }
+      })
+    })
+  })
```

Comment:
We should also add a test suite for `linkWithRedirect`. ;)

Reply:
All done!